server/src/main/java/org/elasticsearch/action/search/SearchTransportService.java

```diff
@@ -91,11 +91,13 @@ public SearchTransportService(
         BiFunction<
             Transport.Connection,
             SearchActionListener<? super SearchPhaseResult>,
-            ActionListener<? super SearchPhaseResult>> responseWrapper
+            ActionListener<? super SearchPhaseResult>> responseWrapper,
+        SearchTransportAPMMetrics searchTransportMetrics
     ) {
         this.transportService = transportService;
         this.client = client;
         this.responseWrapper = responseWrapper;
+        this.searchTransportMetrics = searchTransportMetrics;
```

Comment:
should we add a test that verifies that the instrumentation actually works and the metrics can be retrieved?